scanomatic/ui_server_data/js/som/experiment.js

```diff
@@ -0,0 +1,376 @@
+  }
+  setValidationStatus('#current-fixture', fixtureSelected && activePlates, tooltip);
+  const timeMakesSense = getDurationAsMinutes() >= 0;
+  setValidationStatus(
+    '#project-duration',
+    timeMakesSense,
+    timeMakesSense ? "That's a splendid project duration" : "Nope, you can't reverse time",
+  );
+  InputEnabled(
+    $('#submit-button'),
+    projectPathValid && scannerOK && fixtureSelected && activePlates && timeMakesSense,
+  );
+}
+
+export function setPoetry(input) {
+  $(input).html(poetry).css('font-style', 'italic');
+
+  $(input).on('focus', function handleFocus() {
+    if ($(this).html() === poetry) {
+      $(this).html('').css('font-style', 'normal');
+    }
+  });
+
+  $(input).on('blur', function handleBlur() {
+    if ($(this).html() === '') {
+      $(this).html(poetry).css('font-style', 'italic');
+    }
+  });
+}
+
+export function setExperimentRoot(input) {
+  getPathSuggestions(
+    input,
+    true,
+    '',
+    null,
+    (data) => {
+      projectPath = $(input).val();
+      projectPathValid = data.valid_parent && !data.exists;
+      projectPathInvalidReason = data.reason;
+
+      setValidationStatus(
+        '#project-path',
+        projectPathValid,
+        projectPathValid ? 'Everything is cool' : projectPathInvalidReason,
+      );
+      if (data.valid_parent && !data.exists) {
+        $('#experiment-title').html(`Start Experiment '${data.prefix}'`);
+      } else {
+        $('#experiment-title').html('Start Experiment');
+      }
+    },
+  );
+}
+
+function getDescription(index, description) {
+  return `${"<div class='plate-description'>" +
+        "<input type='hidden' value='"}${index}'>` +
+        `<label for='plate-description-${index}'>Plate ${index}</label>` +
+        `<input class='long' id='plate-description-${index}' value='${description
+        }' placeholder='Enter medium and, if relevant, what experiment this plate is.' onchange='cacheDescription(this);'></div>`;
+}
+
+function setVisiblePlateDescriptions() {
+  const activeIndices = $.map(fixturePlates
+    .filter(value => value.active === true), entry => entry.index);
+
+  const descriptions = $('#plate-descriptions');
+  if (fixturePlates.length === 0) {
+    descriptions.html('<em>Please, select a fixture and plate pinnings first.</em>');
+    return;
+  } descriptions.html('');
+
+  for (let i = 0; i < activeIndices.length; i += 1) {
+    const index = activeIndices[i];
+    descriptions.append(getDescription(
+      index,
+      descriptionCache[index] !== undefined ? descriptionCache[index] : '',
+    ));
+  }
+}
+
+function getPlateSelector(plate) {
+  return (
+    `<div class='pinning'><input type='hidden' value='${plate.index}'>`
+      + `<label for='pinning-plate-${plate.index}'>Plate ${plate.index}</label>`
+      + `<select id='pinning-plate-${plate.index}' class='pinning-selector' onchange='setActivePlate(this); validateExperiment();'>`
```

Comment:
this onchange too

Reply:
basically check all onX occurances

Reply:
I checked all `onclick` yesterday, but forgot about `onchange`. Thanks.

Reply:
Fixed (checked all ` on[a-z]+`)

Reply:
onmouse onkey ondocument perhaps we have others too